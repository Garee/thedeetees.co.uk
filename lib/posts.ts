import fs from "fs";
import path from "path";
import matter from "gray-matter";
import remark from "remark";
import html from "remark-html";

const postsDirectory = path.join(process.cwd(), "posts");

export interface Post {
  id: string;
  date: string;
  title: string;
  body: string;
  thumbnail?: string;
}

export async function getSortedPostsData(): Promise<Post[]> {
  const fileNames: string[] = fs.readdirSync(postsDirectory);

  const allPostsData: Promise<Post>[] = fileNames.map(async (fileName) => {
    const id = fileName.replace(/\.md$/, "");
    return await getPostData(id);
  });

  return Promise.all(allPostsData).then((allPostsData) => {
    return allPostsData.sort(({ date: a }, { date: b }) => {
      if (a < b) {
        return 1;
      } else if (a > b) {
        return -1;
      }
      return 0;
    });
  });
}

export function getAllPostIds(): Array<{ params: { id: string } }> {
  const fileNames: string[] = fs.readdirSync(postsDirectory);
  return fileNames.map((fileName) => {
    return {
      params: {
        id: fileName.replace(/\.md$/, ""),
      },
    };
  });
}

export async function getPostData(id: string): Promise<Post> {
  const fullPath = path.join(postsDirectory, `${id}.md`);
  const fileContents = fs.readFileSync(fullPath, "utf8");

  // Use gray-matter to parse the post metadata section
  const matterResult = matter(fileContents);
  const { date, title, thumbnail } = matterResult.data;

  // Get the post content HTML.
  const contentHtml = await (
    await remark().use(html).process(matterResult.content)
  ).toString();

  return {
    id,
    title,
    thumbnail,
    date: date.toISOString(),
    body: contentHtml,
  };
}
