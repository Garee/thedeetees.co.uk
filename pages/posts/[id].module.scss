.thumbnail {
  max-height: 300px;
  object-fit: cover;
  border-radius: 4px;
  width: 100%;
  height: 100%;
  margin-top: 1rem;
}
