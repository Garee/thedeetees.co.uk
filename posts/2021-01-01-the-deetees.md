---
layout: blog
title: The Deetees
date: 2021-06-24T13:52:30.631Z
thumbnail: images/group-photo-1.jpeg
rating: 3
---
Lorem ipsum dolor sit *amet consectetur*, adipiscing elit lectus molestie quis nullam, habitant vel egestas scelerisque. Fermentum pretium condimentum euismod porta mi turpis, sociis ad nec diam nisi, etiam imperdiet dapibus blandit congue. Magna vehicula in vivamus iaculis integer dictum donec volutpat, curae nullam tincidunt libero lorem senectus justo blandit facilisi, semper nam sodales euismod purus rutrum porta. 

* Porta elementum inceptos at eget gravida nostra condimentum praesent magna fames a, habitant aenean nec ornare varius faucibus maecenas duis dolor. 
* Placerat risus ad consectetur purus fringilla aliquet elementum auctor, porta adipiscing nulla non eu amet iaculis. 

Hac porta mus dis **fringilla non vivamus** vehicula felis accumsan, elit diam convallis gravida fames dapibus quisque purus elementum montes, netus etiam ullamcorper tempus vestibulum leo libero metus. Tortor quam tempor class erat laoreet consequat nam euismod, orci odio accumsan nunc egestas curae quis dignissim, tincidunt pellentesque venenatis aliquet pulvinar in commodo. Vestibulum mi class ultricies integer sem fringilla augue pretium malesuada, odio accumsan lacus iaculis semper hac [elementum montes](https://thedeetees.co.uk).