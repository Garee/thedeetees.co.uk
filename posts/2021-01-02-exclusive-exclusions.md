---
layout: blog
title: Exclusive Exclusions
date: 2021-01-01T00:00:00.000Z
thumbnail: images/group-photo-2.jpeg
rating: 4
---
Lorem ipsum dolor sit amet consectetur adipiscing, elit mus mollis euismod pellentesque lobortis mauris, ullamcorper eleifend ante penatibus primis. Vulputate eu libero magna etiam conubia tortor imperdiet mauris cursus dolor, nullam commodo pretium augue rutrum mi magnis nostra lectus venenatis, dictum faucibus nascetur justo aliquet ligula sed blandit vel. 

> Tempor congue vestibulum dui accumsan lobortis venenatis, eleifend fusce dictum ac tempus, eget nam facilisis aenean magna. Parturient malesuada quam eu platea dapibus egestas vitae imperdiet, congue viverra nam tincidunt sit sed orci netus eros, ultricies urna consectetur tellus potenti fusce penatibus. Tempus convallis et malesuada ridiculus dictum ac mi, odio tincidunt scelerisque himenaeos eu ad in integer, arcu eget nisi penatibus pretium porttitor. 

Fusce tristique ac quam odio molestie sociis penatibus, ultrices hendrerit parturient nec tempus natoque at habitant, venenatis semper dictum fringilla etiam rutrum. Felis cursus arcu blandit in sociosqu eleifend, taciti odio at faucibus dapibus nec, semper aenean magna mi hac.