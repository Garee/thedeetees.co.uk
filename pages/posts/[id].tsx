/* eslint-disable @next/next/no-img-element */

import {
  GetStaticPaths,
  GetStaticPathsContext,
  GetStaticProps,
  GetStaticPropsContext,
} from "next";
import Head from "next/head";
import Link from "next/link";
import Layout from "../../components/layout";
import { getAllPostIds, getPostData, Post as PostData } from "../../lib/posts";
import Date from "../../components/date";
import { appName } from "../../lib/config";
import styles from "./[id].module.scss";
import utilStyles from "../../styles/utils.module.scss";

type PostProps = PostData;

interface ThumbnailProps {
  src?: string;
}

function Thumbnail({ src }: ThumbnailProps) {
  if (!src) {
    return null;
  }

  return <img src={`/${src}`} alt="Post image" className={styles.thumbnail} />;
}

export default function Post({ title, date, body, thumbnail }: PostProps) {
  return (
    <Layout>
      <Head>
        <title>
          {title} | {appName}
        </title>
      </Head>
      <Thumbnail src={thumbnail} />
      <h1 className={utilStyles.headingXl}>{title}</h1>
      <small className={utilStyles.lightText}>
        <Date dateString={date} />
      </small>
      <br />
      <div dangerouslySetInnerHTML={{ __html: body }} />
      <div>
        <Link href="/">
          <a>â¬… Go back</a>
        </Link>
      </div>
    </Layout>
  );
}

export const getStaticProps: GetStaticProps<PostProps> = async ({
  params,
}: GetStaticPropsContext) => {
  let post: PostProps = {
    id: "",
    title: "",
    date: "",
    body: "",
    thumbnail: "",
  };

  if (typeof params?.id === "string") {
    post = await getPostData(params.id);
  }

  return {
    props: post,
  };
};

// The page paths depend on external data.
export const getStaticPaths: GetStaticPaths = async (
  _context: GetStaticPathsContext
) => {
  const paths = getAllPostIds();
  return {
    paths, // Pre-render these at build-time.
    fallback: false, // Other routes should 404.
  };
};
