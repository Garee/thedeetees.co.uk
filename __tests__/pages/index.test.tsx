import renderer from "react-test-renderer";
import posts from "../mocks/posts";
import Index from "../../pages/index";

describe("index page", () => {
  const props = { body: "", posts };
  const index = <Index {...props} />;

  it("renders correctly", () => {
    expect.assertions(1);
    const tree = renderer.create(index).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
