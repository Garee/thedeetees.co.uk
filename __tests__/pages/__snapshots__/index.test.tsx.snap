// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <img
      alt="The Deetees"
      className="borderCircle"
      height={144}
      src="/images/icon.jpg"
      width={144}
    />
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <h1
        className="heading2Xl"
      >
        The Deetees
      </h1>
    </a>
    <nav>
      <a
        href="#about"
      >
        About
      </a>
       / 
      <a
        href="#contact"
      >
        Contact
      </a>
       / 
      <a
        href="#music"
      >
        Music
      </a>
       / 
      <a
        href="#shows"
      >
        Shows
      </a>
    </nav>
  </header>
  <main>
    <section>
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen={true}
        frameBorder="0"
        height="315"
        src="https://www.youtube-nocookie.com/embed/SIUMdVyDD_8"
        style={
          Object {
            "marginTop": "1rem",
          }
        }
        title="YouTube video player"
        width="100%"
      />
    </section>
    <section
      className="headingMd"
      id="about"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </section>
    <section
      id="social"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-evenly",
          }
        }
      >
        <a
          href="https://www.instagram.com/yourdeetees"
        >
          <img
            alt="Instagram"
            height="30"
            src="/social/instagram.svg"
            width="30"
          />
        </a>
        <a
          href="https://www.facebook.com/DeeTeesGlesga"
        >
          <img
            alt="Facebook"
            height="30"
            src="/social/facebook.svg"
            width="30"
          />
        </a>
        <a
          href="https://twitter.com/deetees3"
        >
          <img
            alt="Twitter"
            height="33"
            src="/social/twitter.svg"
            width="33"
          />
        </a>
        <a
          href="https://www.youtube.com/channel/UCohjZ-MlZWw0RZoPvKZkq5A"
        >
          <img
            alt="YouTube"
            height="30"
            src="/social/youtube.svg"
            width="30"
          />
        </a>
      </div>
    </section>
    <section
      id="music"
    >
      <h2>
        Music
      </h2>
    </section>
    <section
      id="shows"
    >
      <h2
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        Shows
      </h2>
    </section>
    <section
      id="posts"
    >
      <h2>
        Latest Posts
      </h2>
      <ul
        className="list"
      >
        <li
          className="listItem"
        >
          <a
            href="/posts/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <h3
              className="headingMd"
            >
              Post #1
            </h3>
          </a>
          <small>
            ðŸ“° 
            <time
              dateTime="2021-01-01"
            >
              1 January, 2021
            </time>
          </small>
        </li>
        <li
          className="listItem"
        >
          <a
            href="/posts/2"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <h3
              className="headingMd"
            >
              Post #2
            </h3>
          </a>
          <small>
            ðŸ“° 
            <time
              dateTime="2021-01-02"
            >
              2 January, 2021
            </time>
          </small>
        </li>
      </ul>
    </section>
    <section
      id="contact"
    >
      <h2>
        Contact
      </h2>
      <p>
        For any enquiries please send us an
         
        <a
          href="mailto:yourdeetees@gmail.com"
        >
          email
        </a>
         or get in touch via social media.
      </p>
    </section>
  </main>
  <footer
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <hr
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <small>
      Â© 
      2021
      , 
      The Deetees
       ðŸŽ¸
    </small>
  </footer>
</div>
`;
